packages/zoe/src/contractSupport/bondingCurves.js

```diff
@@ -35,9 +33,7 @@ export const getCurrentPrice = ({
   const denominator = add(multiply(inputReserve, 10000), inputWithFee);
 
   const outputExtent = floorDivide(numerator, denominator);
-  const newOutputReserve = subtract(outputReserve, outputExtent);
-  const newInputReserve = add(inputReserve, inputExtent);
-  return harden({ outputExtent, newInputReserve, newOutputReserve });
+  return outputExtent;
```

Comment:
Hmm, why would the return value being `outputExtent` mean that `getCurrentPrice` is a bad name? The function name is correct - this function is calculating the current price. The return value is also named appropriately - the current price outcome is an extent that is not the input type - it's the output type. The meaning of both of these didn't change in this PR.

I improved the jsdocs with a return value and made sure the destructured parameters were handled appropriately.

Reply:
I agree that it's calculating the price, but what it returns doesn't seem to be the current price, in the way the old one was. The old output value told how much would be gained and how it would change the underlying reserves, so you could calculate gain and loss.

Maybe I'm making an incorrect distinction, but I think of "current price" as either a unit price ($X per item), or price over amount ($22 for 37 moolah). This function seems to be calculating "how much moolah would I get if I spent 37 simoleans?",  and the return value isn't price, it's the proceeds.  It would look more like a price to me if the question it answered were "How much moola would I have to spend to get 45 Simoleans?"

Reply:
The only reason that newInputReserve and newOutputReserve were returned before was to make reallocation easier, not to calculate gain and loss (these were never sent to the user). Now that the `trade` helper does this, there is no need to do it in `getCurrentPrice`.  There's no difference in behavior here, so I think discussion of whether `getCurrentPrice` is the right name is out of scope for this PR.  I think we should create an issue for this. Uniswap V1 uses `getInputPrice` and `getOutputPrice` as the names for their functions. We could use that terminology and go both directions.

Reply:
The code for uniswap V1 is here: https://github.com/Uniswap/uniswap-v1/blob/c10c08d81d6114f694baa8bd32f555a40f6264da/contracts/uniswap_exchange.vy#L106

Reply:
I'm fine with discussing this in a separate issue. `getInputPrice` paired with `getOutputPrice` doesn't raise the same issues for me as `getCurrentPrice`.

Reply:
Ok, great! Issue is here: https://github.com/Agoric/agoric-sdk/issues/1153